File: src/api/encodings.js
'use strict';

const { errMsg } = require('../utils/app');

function registerEncodingsApi({ app, db }) {
    app.get('/encodings/custom', (req, res) => {
        try {
            return res.json({ ffmpegArgs: db.getCustomEncoding() });
        } catch (err) {
            return res.status(500).json({ error: errMsg(err) });
        }
    });

    app.put('/encodings/custom', (req, res) => {
        try {
            const { ffmpegArgs } = req.body || {};
            if (typeof ffmpegArgs !== 'string') {
                return res.status(400).json({ error: 'ffmpegArgs must be a string' });
            }
            db.setCustomEncoding(ffmpegArgs.trim());
            return res.json({ ffmpegArgs: ffmpegArgs.trim() });
        } catch (err) {
            return res.status(500).json({ error: errMsg(err) });
        }
    });
}

module.exports = { registerEncodingsApi };

File: internal/api/encodings.go
package api

import (
	"net/http"
	"strings"

	"github.com/go-chi/chi/v5"

	"restream/internal/db"
)

// RegisterEncodingsAPI mounts the /encodings/custom endpoints.
func RegisterEncodingsAPI(r chi.Router, database *db.DB) {
	r.Get("/encodings/custom", func(w http.ResponseWriter, r *http.Request) {
		jsonOK(w, map[string]interface{}{"ffmpegArgs": database.GetCustomEncoding()})
	})

	r.Put("/encodings/custom", func(w http.ResponseWriter, r *http.Request) {
		var body struct {
			FFmpegArgs *string `json:"ffmpegArgs"`
		}
		if err := decodeJSON(r, &body); err != nil || body.FFmpegArgs == nil {
			jsonError(w, http.StatusBadRequest, "ffmpegArgs must be a string")
			return
		}
		trimmed := strings.TrimSpace(*body.FFmpegArgs)
		database.SetCustomEncoding(trimmed)
		jsonOK(w, map[string]string{"ffmpegArgs": trimmed})
	})
}
